src/OrleansRuntime/Catalog/Catalog.cs

```diff
@@ -398,7 +398,7 @@ public void UnregisterMessageTarget(ActivationData activation)
             ActivationCollector.TryCancelCollection(activation);
             activationsDestroyed.Increment();
 
-            scheduler.UnregisterWorkContext(new SchedulingContext(activation));
+            scheduler.UnregisterWorkContext(activation.SchedulingContext);
```

Comment:
Fixed, thanks. Now of the 3 `SchedulingContext` ctors, 2 are used once (`SystemTarget` & `ActivationData`) and one is unused (for 'system thread').

I could remove the latter for hygiene, but I rather leave it for now as it looks like it may prove itself useful soon.